### ISSUE TITLE
platform.native plugin incompatible with Gradle 5

### ISSUE THREAD

**Issue by trevjonez** [JetBrains/kotlin-native#2642]
Lockable set property was removed in: https://github.com/gradle/gradle/pull/7371 so I assume gradle 5.0 and up given the merge date.

Build scan of observed issue: https://scans.gradle.com/s/54l4oq6rr73q4 and gradle 5.2
```
org.gradle.api.plugins.InvalidPluginException: An exception occurred applying plugin request [id: 'org.jetbrains.kotlin.platform.native', version: '1.3.21']
  Caused by: org.gradle.api.internal.plugins.PluginApplicationException: Failed to apply plugin [class 'org.jetbrains.kotlin.gradle.plugin.experimental.plugins.KotlinNativePlugin']
    Caused by: org.gradle.api.reflect.ObjectInstantiationException: Could not create an instance of type org.jetbrains.kotlin.gradle.plugin.experimental.internal.KotlinNativeMainComponent
      Caused by: org.gradle.internal.instantiation.ClassGenerationException: Could not generate a decorated class for class org.jetbrains.kotlin.gradle.plugin.experimental.internal.KotlinNativeMainComponent.
        Caused by: java.lang.NoClassDefFoundError: org/gradle/api/internal/provider/LockableSetProperty
          Caused by: java.lang.ClassNotFoundException: org.gradle.api.internal.provider.LockableSetProperty
```

Possible overlap with #2440

**Comment by ilmat192** [comment]
Thank you for the report. Yes, it's a known problem and we address it. Closing as a duplicate of #2440.

**Comment by hadyang** [comment]
Same issuse.

My gradle is 5.2.1, [#2440](https://github.com/JetBrains/kotlin-native/issues/2440) not fix this

```
Caused by: java.lang.NoClassDefFoundError: org/gradle/api/internal/provider/LockableSetProperty
	at org.gradle.internal.reflect.ClassInspector.inspectClass(ClassInspector.java:69)
	at org.gradle.internal.reflect.ClassInspector.visitGraph(ClassInspector.java:55)
	at org.gradle.internal.reflect.ClassInspector.inspect(ClassInspector.java:35)
	at org.gradle.internal.instantiation.AbstractClassGenerator.inspectType(AbstractClassGenerator.java:199)
	at org.gradle.internal.instantiation.AbstractClassGenerator.generateUnderLock(AbstractClassGenerator.java:152)
	... 176 more
Caused by: java.lang.ClassNotFoundException: org.gradle.api.internal.provider.LockableSetProperty
	... 181 more
```

**Comment by ilmat192** [comment]
The fix is included in `1.3.30-eap-11`, not in `1.3.21`. Note however that the plugin is still incompatible with Gradle 5.2 and making it compatible breaks compatibility with earlier versions.
